package Chapter3_Assertions_and_Exceptions.Questions;

import java.io.IOException;
import java.sql.SQLException;

public class Question3 {
//    public static void main(String[] args) {
//        Question3 a = new Question3();
//        try {
//            a.blowUp();
//            System.out.print("a");
//        } catch (IOException e | SQLException e) {    // -- Can't do this
//            System.out.print("c");
//        } finally {
//            System.out.print("d");
//        }
//    }
//    void blowUp() throws IOException, SQLException {
//        throw new SQLException();
//    }
}

// What is the result?

// A. ad
// B. acd
// C. cd
// D. d
// E. Compilation fails                 --> CORRECT
// F. An exception is thrown at runtime